geo-traits/src/polygon.rs

```diff
@@ -0,0 +1,134 @@
+use std::marker::PhantomData;
+
+use crate::iterator::PolygonInteriorIterator;
+use crate::line_string::UnimplementedLineString;
+use crate::{Dimensions, LineStringTrait};
+use geo_types::{CoordNum, LineString, Polygon};
+
+/// A trait for accessing data from a generic Polygon.
+///
+/// A `Polygon`’s outer boundary (_exterior ring_) is represented by a
+/// [`LineString`][LineStringTrait]. It may contain zero or more holes (_interior rings_), also
+/// represented by `LineString`s.
+///
+/// Refer to [geo_types::Polygon] for information about semantics and validity.
+pub trait PolygonTrait: Sized {
+    /// The coordinate type of this geometry
+    type T: CoordNum;
+
+    /// The type of each underlying ring, which implements [LineStringTrait]
+    type RingType<'a>: 'a + LineStringTrait<T = Self::T>
+    where
+        Self: 'a;
+
+    /// The dimension of this geometry
+    fn dim(&self) -> Dimensions;
+
+    /// The exterior ring of the polygon
+    fn exterior(&self) -> Option<Self::RingType<'_>>;
```

Comment:
Sorry if I missed the discussion on this, but under what situations would the exterior ring be `None`?

Reply:
When it's `POLYGON EMPTY`

See some discussion: https://github.com/georust/geo/pull/1157#discussion_r1797484262